lib/srv/discovery/fetchers/db/azure_dbserver.go

```diff
@@ -81,3 +80,7 @@ func (p *azureDBServerPlugin) NewDatabaseFromServer(server *azure.DBServer, log
 	}
 	return database
 }
+
+func (f *azureDBServerPlugin) MatchesResource(db types.Database) bool {
+	return db.GetType() == types.DatabaseTypeAzure
```

Comment:
sorry about the confusing fetchers. 

azureDBServerPlugin is shared for MySQL and PostgreSQL. We might have to split `azureDBServerPlugin` into two or maybe add a `protocol` string through constructor to remember the type. 

then
```suggestion
	return db.GetType() == types.DatabaseTypeAzure && p.protocol == db.GetProtocol()
```